test/unit/org/apache/cassandra/cql3/CQLTester.java

```diff
@@ -462,54 +466,45 @@ protected static void reinitializeNetwork(Consumer<Cluster.Builder> clusterConfi
         clusters.clear();
         sessions.clear();
 
-        initializeNetwork(server -> {}, clusterConfigurator);
+        clusterBuilderConfigurator = clusterConfigurator;
+
+        startServer(serverConfigurator);
     }
 
-    private static void initializeNetwork(Consumer<Server.Builder> decorator, Consumer<Cluster.Builder> clusterConfigurator)
+    private static void startServer(Consumer<Server.Builder> decorator)
     {
-        startServer(decorator);
-
-        for (ProtocolVersion version : PROTOCOL_VERSIONS)
-        {
-            if (clusters.containsKey(version))
-                continue;
+        Server.Builder serverBuilder = new Server.Builder().withHost(nativeAddr).withPort(nativePort);
+        decorator.accept(serverBuilder);
+        server = serverBuilder.build();
+        ClientMetrics.instance.init(Collections.singleton(server));
+        server.start();
+    }
 
-            SocketOptions socketOptions = new SocketOptions()
-                                          .setConnectTimeoutMillis(Integer.getInteger("cassandra.test.driver.connection_timeout_ms", DEFAULT_CONNECT_TIMEOUT_MILLIS)) // default is 5000
-                                          .setReadTimeoutMillis(Integer.getInteger("cassandra.test.driver.read_timeout_ms", DEFAULT_READ_TIMEOUT_MILLIS)); // default is 12000
+    private static Cluster initClientCluster(ProtocolVersion version)
+    {
+        SocketOptions socketOptions = 
+                new SocketOptions().setConnectTimeoutMillis(Integer.getInteger("cassandra.test.driver.connection_timeout_ms",
+                                                            DEFAULT_CONNECT_TIMEOUT_MILLIS)) // default is 5000
+                                   .setReadTimeoutMillis(Integer.getInteger("cassandra.test.driver.read_timeout_ms",
+                                                         DEFAULT_READ_TIMEOUT_MILLIS)); // default is 12000
 
-            logger.info("Timeouts: {} / {}", socketOptions.getConnectTimeoutMillis(), socketOptions.getReadTimeoutMillis());
+        logger.info("Timeouts: {} / {}", socketOptions.getConnectTimeoutMillis(), socketOptions.getReadTimeoutMillis());
 
-            Cluster.Builder builder = Cluster.builder()
-                                             .withoutJMXReporting()
-                                             .addContactPoints(nativeAddr)
-                                             .withClusterName("Test Cluster")
-                                             .withPort(nativePort)
-                                             .withSocketOptions(socketOptions);
+        Cluster.Builder builder = Cluster.builder().withoutJMXReporting().addContactPoints(nativeAddr)
+                .withClusterName("Test Cluster").withPort(nativePort).withSocketOptions(socketOptions);
```

Comment:
Nit: maybe the previous on-column indentation was a bit easier to read:
```suggestion
        Cluster.Builder builder = Cluster.builder()
                                         .withoutJMXReporting()
                                         .addContactPoints(nativeAddr)
                                         .withClusterName("Test Cluster")
                                         .withPort(nativePort)
                                         .withSocketOptions(socketOptions);
```